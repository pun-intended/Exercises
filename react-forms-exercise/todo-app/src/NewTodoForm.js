import React, { useState } from 'react'
import {v4 as uuid} from "uuid"
const INITIAL_STATE = {
    task: ""
}

function NewTodoForm({addTodo}){
    const [formData, setFormData] = useState(INITIAL_STATE)
    const handleSubmit = (evt) => {
        evt.preventDefault()
        addTodo(formData)
        setFormData(INITIAL_STATE)
    }

    const handleChange = evt => {
        const { name, value } = evt.target
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    }

    return(
        <div>
            <form onSubmit={handleSubmit}>
                <label htmlFor="task">Task</label>
                <input name="task" value={formData.task} onChange={handleChange}></input>
                <button>Add</button>
            </form>
        </div>
    )
}

export default NewTodoForm