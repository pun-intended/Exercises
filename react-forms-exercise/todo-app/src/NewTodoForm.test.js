import { render, screen } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// Smoke test
render(<NewTodoForm />)

// Snapshot test
const {asFragment} = render(<NewTodoForm />)
expect(asFragment()).toMatchSnapshot()