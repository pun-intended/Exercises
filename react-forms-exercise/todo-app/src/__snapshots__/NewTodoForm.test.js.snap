// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task
      </label>
      <input
        name="task"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
